app/src/main/java/org/schabi/newpipe/settings/ContentSettingsFragment.java

```diff
@@ -195,6 +197,33 @@ public void onActivityResult(final int requestCode, final int resultCode,
         }
     }
 
+    private void showExportDialog(final Context c, final String path) {
+        final EditText fileNameET = new EditText(c);
+
+        final SimpleDateFormat sdf = new SimpleDateFormat("yyyyMMdd_HHmmss", Locale.US);
+        fileNameET.setText("NewPipeData-" + sdf.format(new Date()) + ".zip");
+
+        fileNameET.setInputType(InputType.TYPE_CLASS_TEXT | InputType.TYPE_TEXT_VARIATION_URI);
+        fileNameET.setHint(R.string.export_data_file_name_hint);
+
+        final AlertDialog dialog = new AlertDialog.Builder(c)
+                .setTitle(R.string.export_data_file_name)
+                .setIcon(R.drawable.ic_import_export)
+                .setNegativeButton(R.string.cancel, null)
+                .setPositiveButton(R.string.export_button, (dialog1, which) -> {
+                    final String fileName = fileNameET.getText().toString();
+                    if (FilenameUtils.isValidFileName(fileName, "zip")) {
```

Comment:
We could also append ".zip" when the file name does not have that extension.

Reply:
So, should I don't ask extension of file in file name or if user enter file name without any dots, I append ".zip" at the end of it?

What should I do if user enter "foo.rar" ? make it "foo.zip"? or "foo.rar.zip"?